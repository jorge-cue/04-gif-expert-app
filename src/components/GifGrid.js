import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
// import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>
            <div className="card-grid">
            
            { loading && <p>Loading...</p> }
            {
                images.map( gif => ( 
                    <GifGridItem 
                        key= {gif.id} 
                        { ...gif } 
                    /> ))
            }
            </div>
        </>
    )
}
