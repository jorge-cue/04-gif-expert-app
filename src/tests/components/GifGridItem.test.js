import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";


describe('GifGridItem tests suite', () => { 
    const image = {
        id: 'Ignored id',
        title: 'Test Image Title',
        url: 'https://test_image_url',
    }
    const wrapper = shallow( <GifGridItem  { ...image } /> );

    test('shoud render the component successfully', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should use title and url in proper properties of img', () => {
        const imgElement = wrapper.find('img');
        expect(imgElement.prop('alt')).toBe(image.title);
        expect(imgElement.prop('src')).toBe(image.url);
    });

    test('should use title un paragraph', () => {
        const pElement = wrapper.find('p');
        expect(pElement.text().trim()).toBe(image.title);
    })

    test('should use "card" in className for div', () => {
        const divElement = wrapper.find('div');
        const className = 'card';
        expect(divElement.prop('className')).toContain(className);        
    })
    
    // test('should use "animate__fadeIn" in className for div', () => {
    //     const divElement = wrapper.find('div');
    //     const className = 'animate__fadeIn';
    //     expect(divElement.prop('className')).toContain(className);        
    // }) 
});