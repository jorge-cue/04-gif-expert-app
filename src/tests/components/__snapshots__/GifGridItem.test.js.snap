// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests suite shoud render the component successfully 1`] = `
<div
  className="card"
>
  <img
    alt="Test Image Title"
    src="https://test_image_url"
  />
  <p>
     
    Test Image Title
     
  </p>
</div>
`;
